Tests/test_lib_image.py

```diff
@@ -30,3 +60,69 @@ def test_setmode():
         im.im.setmode("L")
     with pytest.raises(ValueError):
         im.im.setmode("RGBABCDE")
+
+
+@pytest.mark.parametrize("mode", mode_names_not_bgr)
+def test_equal(mode):
+    num_img_bytes = len(Image.new(mode, (2, 2)).tobytes())
+    # alternatively, random.randbytes() in Python 3.9
+    data = secrets.token_bytes(num_img_bytes)
+    img_a = Image.frombytes(mode, (2, 2), data)
+    img_b = Image.frombytes(mode, (2, 2), data)
+    assert img_a.tobytes() == img_b.tobytes()
+    assert img_a.im == img_b.im
+
+
+# With mode "1" different bytes can map to the same value,
+# so we have to be more specific with the values we use.
+def test_not_equal_mode_1():
+    data_a = data_b = bytes(secrets.choice(b"\x00\xff") for i in range(4))
+    while data_a == data_b:
+        data_b = bytes(secrets.choice(b"\x00\xff") for i in range(4))
+    # We need to use rawmode "1;8" so that each full byte is interpreted as a value
+    # instead of the bits in the bytes being interpreted as values.
+    img_a = Image.frombytes("1", (2, 2), data_a, "raw", "1;8")
+    img_b = Image.frombytes("1", (2, 2), data_b, "raw", "1;8")
+    assert img_a.tobytes() != img_b.tobytes()
+    assert img_a.im != img_b.im
+
+
+@pytest.mark.parametrize("mode", [mode for mode in mode_names_not_bgr if mode != "1"])
+def test_not_equal(mode):
+    num_img_bytes = len(Image.new(mode, (2, 2)).tobytes())
+    # alternatively, random.randbytes() in Python 3.9
+    data_a = data_b = secrets.token_bytes(num_img_bytes)
+    while data_a == data_b:
+        data_b = secrets.token_bytes(num_img_bytes)
+    img_a = Image.frombytes(mode, (2, 2), data_a)
+    img_b = Image.frombytes(mode, (2, 2), data_b)
+    assert img_a.tobytes() != img_b.tobytes()
+    assert img_a.im != img_b.im
+
+
+@pytest.mark.skip(reason="no way to directly set C bytes from Python")
```

Comment:
If this test is never run, then why add it?

Reply:
I thought someone else might know a way to do it. If I change to code to always use 0xffffffff for the mask, "test_image_reduce.py::test_mode_LA_opaque[1]" fails. So there are situations where identical images can end up with different values in their unused bytes, and that should be tested. I just don't know how.

Reply:
Thanks. That works for RGB and YCbCr, but HSV, LAB, LA, La, and PA don't have a 4-byte raw mode.